django_fsm_log/checks.py

```diff
@@ -0,0 +1,71 @@
+import ast
+import inspect
+
+import django
+from django.core.checks import (
+    Warning,
+    register,
+    Tags,
+)
+from django.core.exceptions import FieldDoesNotExist
+
+from django_fsm_log.models import StateLog
+
+
+@register(Tags.compatibility)
+def integer_object_id_check(app_configs, **kwargs):
+    errors = []
+    for app in django.apps.apps.get_app_configs():
+
+        # Skip third party apps.
+        if app.path.find('site-packages') > -1:
+            continue
+
+        for model in app.get_models():
+            for check_message in check_model_for_integer_object_id(model):
+                errors.append(check_message)
+    return errors
+
+
+def check_model_for_integer_object_id(model):
+    """Check a single model.
+
+    Yields (django.checks.CheckMessage)
+    """
+    model_source = inspect.getsource(model)
+    model_node = ast.parse(model_source)
+
+    for node in model_node.body[0].body:
+
+        # Check if node is a model field.
+        if not isinstance(node, ast.Assign):
+            continue
+
+        if len(node.targets) != 1:
+            continue
+
+        if not isinstance(node.targets[0], ast.Name):
+            continue
+
+        field_name = node.targets[0].id
+        try:
+            field = model._meta.get_field(field_name)
+        except FieldDoesNotExist:
+            continue
+
+        # Node is a model field here
+
+        # Check if field has foreign key to StateLog defined
+        for kw in node.value.keywords:
+            if kw.arg == 'to':
+                to = kw
+                if to == 'django_fsm_log.StateLog' or isinstance(to, StateLog):
+                    yield Warning(
+                        'StateLog has changed its object_id from PositiveIntegerField to TextField to better support '
+                        'the primary key types in Django.',
+                        hint='Migration PositiveIntegerField -> TextField is potentially needed on field {}.'.format(
+                            field.name
+                        ),
+                        obj=field,
+                        id='django_fsm_log.W001',
+                    )
```

Comment:
Is this warning absolutely necessary?  I'm assuming it relates to https://code.djangoproject.com/ticket/25012 or similar?

Reply:
Basically yeah, one of the only solid bits of contention in the issue where this was discussed was surrounding the use of the field by user based code which implicitly expected an integer.
I am also aware that I've been tripped up by types of foreign key references changing in the past in my own code and it's within the scope of the check framework to automate looking for these kinds of errors and warning users about them. 
We can also nerf it again at some point in the futurewhen we feel we've had enough of a deprecation period. I just wish there was a way I could also check to see if any python code references these fields.